qiskit/qobj/qasm_qobj.py

```diff
@@ -368,7 +368,32 @@ class QasmQobjExperimentHeader(QobjDictField):
 
 class QasmQobjExperimentConfig(QobjDictField):
     """Configuration for a single QASM experiment in the qobj."""
-    pass
+
+    def __init__(self, calibrations=None, **kwargs):
+        """
+        Args:
+            calibrations (QasmQobjCalibrations): Information required for
+                                                 Pulse gates.
+            kwargs: Additional free form key value fields to add to the
+                configuration.
+        """
+        if calibrations:
+            self.calibrations = calibrations
+        super().__init__(**kwargs)
+
+
+class QasmQobjCalibrations(QobjDictField):
+    """The description of all the gate calibrations. Each gate dictionary should contain
+    "name", "qubits", "params", and "instructions" fields."""
+
+    def __init__(self, gates, pulse_library):
+        """
+        Args:
+            gates (list): A list of dictionaries which specify a gate by name, qubits,
```

Comment:
What does gates look like? Should this have a `Qobj` object in a similar way to https://github.com/Qiskit/qiskit-terra/blob/4f30d98f6a4ad3bb3f8b5869a910ccea2c50fdb4/qiskit/providers/models/pulsedefaults.py#L99. I see these as better `CmdDef` and we should possibly considering upgrading this definition as well (although separately as this would be a breaking change).

Reply:
I think the schema shows it more clearly, or the PR summary. It's possible I made some mistake here in the class

Reply:
I agree the schema makes it more clear. The pattern followed by the rest of the Qobj object model is to provide an object to represent this structure.

Reply:
If the fields here are required (which looking at the schema they are) then we should probably be defining these somewhere in the object model. Personally I feel like the `__init__` should just check the keys on the input dictionary. 

@taalexander is correct though the object model that is here for everything else would expect a `CalibrationEntry` class definition that this would be a list of. But, this is just a leftover from how marshmallow worked (we had to keep it for backwards compat) but moving forward I don't feel we need to continue this model now that marshmallow has been removed.But I think we should still validate the fields since they're strictly defined.